packages/noya-state/src/reducers/styleReducer.ts

```diff
@@ -280,6 +289,49 @@ export function styleReducer(
         draft.borders[index].position = position;
       });
     }
+    case 'setFillType': {
+      const [, index, type] = action;
+
+      return produce(state, (draft) => {
+        if (!draft.fills || !draft.fills[index]) return;
+        draft.fills[index].fillType = type;
+
+        if (!draft.fills[index].gradient) {
+          draft.fills[index].gradient = Models.fill.gradient;
+        }
+      });
+    }
+    case 'setGradientColor': {
+      const [, fillIndex, gradientIndex, position, color] = action;
+
+      return produce(state, (draft) => {
+        if (!draft.fills || !draft.fills[fillIndex]) return;
+        if (!draft.fills[fillIndex].gradient) return;
+
+        if (gradientIndex !== -1) {
+          draft.fills[fillIndex].gradient.stops[
+            gradientIndex
+          ].position = position;
+          draft.fills[fillIndex].gradient.stops[gradientIndex].color = color;
+        } else {
+          draft.fills[fillIndex].gradient.stops.push({
```

Comment:
Maybe should sort the gradient stops whenever we push or modify their position?

Reply:
I have an issue with this 😅 I did used to do this but since I am storing the index of the current stop I am modifying when I resort the array this index changes but I am not 100% sure where in the code I should do this index change 🤔

Reply:
That's a good point, I didn't think of that. Maybe the way you have it now is good then, although I will probably need to add a sort to the rendering code at some point